src/ui/views/session/discover/DiscoverAdmin.tsx

```diff
@@ -0,0 +1,544 @@
+import { Label } from "../../../atoms/text/Label";
+import { Text } from "../../../atoms/text/Text";
+import { useAsyncCallback } from "../../../hooks/useAsyncCallback";
+import { useHydrogen } from "../../../hooks/useHydrogen";
+import { RoomTypes, useRoomsOfType } from "../../../hooks/useRoomsOfType";
+import { getAvatarHttpUrl, getIdentifierColorNumber } from "../../../utils/avatar";
+import { useFeaturedRooms } from "../../components/FeaturedRoomsProvider";
+import { RoomSummaryProvider } from "../../components/RoomSummaryProvider";
+import { SettingTile } from "../../components/setting-tile/SettingTile";
+import ChevronBottomIC from ".././../.././../../res/ic/chevron-bottom.svg";
+import ChevronTopIC from ".././../.././../../res/ic/chevron-top.svg";
+import BinIC from ".././../.././../../res/ic/bin.svg";
+import "./DiscoverAdmin.css";
+import { RepositoryEvents } from "./DiscoverView";
+import { SceneData, sceneDataToScene, SceneSubmission } from "./SceneSubmission";
+import { useFeaturedWorlds } from "../../components/FeaturedWorldsProvider";
+import { useFeaturedScenes } from "../../components/FeaturedScenesProvider";
+import { Input } from "../../../atoms/input/Input";
+import { eventByOrderKey, isValidRoomId } from "../../../utils/matrixUtils";
+import { getRoomSummary } from "../../../hooks/useRoomSummary";
+import { useOrderString } from "../../../hooks/useOrderString";
+import { useOrderMove } from "../../../hooks/useOrderMove";
+
+interface FeaturedItemProps {
+  before?: ReactNode;
+  children?: ReactNode;
+  after?: ReactNode;
+}
+function FeaturedItem({ before, children, after }: FeaturedItemProps) {
+  return (
+    <div className="FeaturedItem flex items-center gap-md">
+      {before && <div>{before}</div>}
+      <div className="grow">{children}</div>
+      {after && <div>{after}</div>}
+    </div>
+  );
+}
+
+function FeatureRoom({ room }: { room: Room }) {
+  const { session, platform } = useHydrogen(true);
+  const featuredRooms = useFeaturedRooms(room);
+  const [allRooms] = useRoomsOfType(session, RoomTypes.Room);
+  const unFeaturedRoom = allRooms.filter(
+    (room) => !featuredRooms.find((stateEvent) => stateEvent.state_key === room.id)
+  );
+  const formRef = useRef<HTMLFormElement>(null);
+  const orderedRooms = featuredRooms.sort(eventByOrderKey);
+  const order = useOrderString();
+  const handleMoveUp = useOrderMove(session, room.id, orderedRooms, order);
+
+  const {
+    callback: featureRoom,
+    loading,
+    error,
+  } = useAsyncCallback(
+    async (roomId: string) => {
+      if (!isValidRoomId(roomId)) throw Error("Invalid roomId");
+      const response = await getRoomSummary(session, roomId);
+      const summaryData = await response.json();
+      if (!summaryData.room_id) throw Error("Can not feature room. Either room is private or does not exist.");
+      const firstEvent = orderedRooms[0];
+      const firstOrder = firstEvent.content.order;
```

Comment:
I think this is because `firstEvent` can be `undefined`. Wondering why ts is not saying that?